mtags-interfaces/src/main/java/scala/meta/pc/PresentationCompiler.java

```diff
@@ -63,6 +63,11 @@
      */
     public abstract CompletableFuture<DefinitionResult> definition(OffsetParams params);
 
+    /**
+     * Return the necessary imports for a symbol at the given position.
+     */
+    public abstract CompletableFuture<List<AutoImportsResult>> autoImports(String name, OffsetParams params);
```

Comment:
Maybe we could do something more general:
```
public abstract CompletableFuture<List<CodeActionResult>> codeAction(String input, ActionType actionType, OffsetParams params);

enum ActionType{
  Import, Implement, TypeAnnotation
}
```
otherwise we will be adding a lot of different methods for each action.

Reply:
Good point: I'm torn about this. From one end, I share your concern, from another I'm not sure of what code actions will be implemented so I'm wary of generalizing the API without knowing the use cases.

Reply:
Ok, let's leave it then and see if generalizing it will be useful in the future.